test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -144,31 +145,51 @@
-    (tk-testutils/with-app-with-config
+    (let [config (jruby-testutils/jruby-config {:max-active-instances 4
+                                                :borrow-timeout default-borrow-timeout})]
+      (tk-testutils/with-app-with-config
+       app
+       [pool-manager/jruby-pool-manager-service]
+       (jruby-testutils/jruby-tk-config config)
+       (let [pool-manager-service (tk-app/get-service app :PoolManagerService)
+             pool-context (pool-manager-protocol/create-pool pool-manager-service config)]
+         ;; set a ruby constant in each instance so that we can recognize them
+         (is (true? (set-constants-and-verify pool-context 4)))
+         (pool-manager-protocol/flush-pool! pool-manager-service pool-context)
+         (is (true? (timed-await (:pool-agent pool-context)))
+             (str "timed out waiting for the flush to complete, stack:\n"
+                  (get-all-stack-traces-as-str)))
+         ;; now the pool is flushed, so the constants should be cleared
+         (is (true? (verify-no-constants pool-context 4))))))))
+
+(deftest ^:integration flush-jruby-pool-for-shutdown-with-pool-manager-test
+  (testing "Flushing the pool for shutdown results in no JRubyInstances left"
+    (let [config (jruby-testutils/jruby-config {:max-active-instances 4
+                                                :borrow-timeout default-borrow-timeout})]
+      (tk-testutils/with-app-with-config
       app
-      [jruby/jruby-pooled-service]
-      (jruby-testutils/jruby-tk-config
-       (jruby-testutils/jruby-config {:max-active-instances      4
-                                             :borrow-timeout default-borrow-timeout}))
-      (let [jruby-service (tk-app/get-service app :JRubyService)
-            context (tk-services/service-context jruby-service)
-            pool-context (:pool-context context)]
-        ;; set a ruby constant in each instance so that we can recognize them
-        (is (true? (set-constants-and-verify pool-context 4)))
-        (jruby-protocol/flush-jruby-pool! jruby-service)
-        (is (true? (timed-await (:pool-agent pool-context)))
-            (str "timed out waiting for the flush to complete, stack:\n"
-                 (get-all-stack-traces-as-str)))
-        ;; now the pool is flushed, so the constants should be cleared
-        (is (true? (verify-no-constants pool-context 4)))))))
+      [pool-manager/jruby-pool-manager-service]
+      (jruby-testutils/jruby-tk-config config)
+      (let [pool-manager-service (tk-app/get-service app :PoolManagerService)
+            pool-context (pool-manager-protocol/create-pool pool-manager-service config)]
+        ;; wait for all jrubies to be added to the pool
+        (jruby-testutils/wait-for-jrubies-from-pool-context pool-context)
+        (is (= 4 (.size (jruby-core/get-pool pool-context))))
+        (pool-manager-protocol/flush-pool-for-shutdown! pool-manager-service pool-context)
```

Comment:
It reduces the logic that a user of the service would have to write - instead of creating the promise and dereferencing it, they can just call the service function... assuming they want it to be synchronous, as the jruby service has it. I couldn't think of any reason why you wouldn't want that call to be synchronous, but I also don't know the backstory for why the service stop was that way.